app/services/support_interface/tad_degree_class_export.rb

```diff
@@ -0,0 +1,91 @@
+    end
+
+    def call(*)
+      report = choices_with_courses_and_subjects.find_each.map do |choice|
+        next if choice.phase == 'apply_2' && !choice.is_latest_a2_app
+
+        subject = MinisterialReport.determine_dominant_course_subject_for_report(choice.course_name, choice.course_level, choice.subject_names.zip(choice.subject_codes).to_h)
+
+        {
+          subject: subject,
+          route: choice.route,
+          degree_class: choice.degree_class,
+          applications: 0,
+          offers_received: 0,
+          number_of_acceptances: 0,
+          number_of_declined_applications: 0,
+          number_of_rejected_applications: 0,
+          number_of_withdrawn_applications: 0,
+          course_name: choice.course_name,
+          course_level: choice.course_level,
+          subject_names: choice.subject_names,
+          subject_codes: choice.subject_codes,
+          status: choice.status,
+        }
+      end
+
+      report = sort_and_group_by_subject_route_and_grade(report)
+
+      report.map do |key, ary|
+        ary.each do |hash|
+          MinisterialReport::DEGREE_CLASS_REPORT_STATUS_MAPPING[hash[:status].to_sym].each do |mapped_status|
+            report[key].first[mapped_status] += 1
+          end
+        end
+      end
+
+      report = report.values.flatten.each do |row|
+        row.delete(:course_name)
+        row.delete(:course_level)
+        row.delete(:subject_names)
+        row.delete(:subject_codes)
+        row.delete(:status)
+      end
+
+      report.reject! do |row|
+        row[:applications].zero? &&
+          row[:offers_received].zero? &&
+          row[:number_of_acceptances].zero? &&
+          row[:number_of_declined_applications].zero? &&
+          row[:number_of_rejected_applications].zero? &&
+          row[:number_of_withdrawn_applications].zero?
+      end
+
+      report
+    end
+
+    alias data_for_export call
+
+  private
+
+    def sort_and_group_by_subject_route_and_grade(report)
+      report = report.compact.uniq.flatten.sort_by { |row| row[:subject] }
+      report.group_by { |hash| grouping_key(hash) }
+    end
+
+    def grouping_key(hash)
+      hash[:subject].to_s + hash[:route].to_s + hash[:degree_class].to_s
+    end
+
+    def choices_with_courses_and_subjects
+      ApplicationChoice
+        .select('application_choices.id as id, application_choices.status as status, providers.provider_type as route, application_qualifications.grade as degree_class, application_form.id as application_form_id, application_form.phase as phase, courses.name as course_name, courses.level as course_level, ARRAY_AGG(subjects.name) as subject_names, ARRAY_AGG(subjects.code) as subject_codes, (CASE WHEN a2_latest_application_forms.candidate_id IS NOT NULL THEN true ELSE false END) AS is_latest_a2_app')
+        .joins(application_form: :application_qualifications)
+        .joins(course_option: { course: :provider })
+        .joins(course_option: { course: :subjects })
```

Comment:
What happens if a course has more than one subject? (I'm not sure I've handled this myself in the other PR).

Reply:
this [method](https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/initializers/ministerial_report.rb#L148) is used on line 15 to determine what is considered the main/dominant subject based on the TAD business rules